core/controllers/admin.py

```diff
@@ -253,13 +253,32 @@ def post(self):
             elif self.payload.get('action') == 'update_feature_flag_rules':
                 feature_name = self.payload.get('feature_name')
                 new_rule_dicts = self.payload.get('new_rules')
-                commit_message = self.payload.get('message')
+                commit_message = self.payload.get('commit_message')
+                if not isinstance(feature_name, python_utils.BASESTRING):
+                    raise self.InvalidInputException(
+                        'feature_name should be string, got \'%s\'.' % (
+                            feature_name))
+                elif not isinstance(commit_message, python_utils.BASESTRING):
+                    raise self.InvalidInputException(
+                        'commit_message should be string, got \'%s\'.' % (
+                            commit_message))
+                elif (not isinstance(new_rule_dicts, list) or not all(
+                        [isinstance(rule_dict, dict)
+                         for rule_dict in new_rule_dicts])):
+                    raise self.InvalidInputException(
+                        'new_rules should be a list of dicts, got \'%s\'.' % (
+                            new_rule_dicts))
                 try:
                     platform_feature_services.update_feature_flag_rules(
                         feature_name, self.user_id, commit_message,
                         new_rule_dicts)
-                except Exception as e:
+                except utils.ValidationError as e:
                     raise self.InvalidInputException(e)
+                except Exception as e:
+                    if 'Unknown feature flag' in python_utils.UNICODE(e):
```

Comment:
Could we use a specific exception type for this? Otherwise, this seems a bit fragile. If we change the error message for the unknown feature flag scenario, this check will break.

Reply:
Done, added `FeatureFlagNotFoundException`.